es-ds-components/components/es-dropdown-select.vue

```diff
@@ -33,7 +33,10 @@ const isInputClicked = ref(false);
 const isLabelClicked = ref(false);
 const isDisabled = computed(() => props.disabled || props.options.length === 0);
 
-const isSelected = (option: any) => option === props.modelValue;
+const isSelected = (option: any) => {
+    const compareOption = props.options.length > 0 && typeof props.options[0] === 'object' ? option.value : option;
```

Comment:
Nit: instead of `props.options.length > 0 && typeof props.options[0] === 'object'`, would it be simpler to say `typeof option === 'object'`?